ibl_to_nwb/AlyxToNWB/alyx_to_nwb_converter.py

```diff
@@ -296,10 +290,22 @@ def create_timeseries_ecephys(self):
                     timestamps_names = self.one_data.data_attrs_dump['_iblqc_ephysTimeRms.timestamps']
                     data_names = self.one_data.data_attrs_dump['_iblqc_ephysTimeRms.rms']
                     for data_idx, data in enumerate(i['data']):
-                        mod.add(TimeSeries(name=data_names[data_idx],
-                                           description=i['description'],
-                                           timestamps=i['timestamps'][timestamps_names.index(data_names[data_idx])],
-                                           data=data))
+                        probe_no = [j for j in range(self.no_probes)
+                                    if self.nwb_metadata['Probes'][j]['name'] in data_names[data_idx]][0]
+                        if data.shape[1] > self.one_data.data_attrs_dump['electrode_table_length'][probe_no]:
+                            if 'channels.rawInd' in self.one_data.loaded_datasets:
+                                channel_idx = self.one_data.loaded_datasets['channels.rawInd'][probe_no].data
+                            else:
+                                warnings.warn('could not find channels.rawInd')
+                                break
+                        else:
+                            channel_idx = np.array(range(data.shape[1]))
+                        mod.add(ElectricalSeries(name=data_names[data_idx],
+                                                 description=i['description'],
+                                                 timestamps=i['timestamps'][
+                                                     timestamps_names.index(data_names[data_idx])],
+                                                 data=data[:, channel_idx],
```

Comment:
Wouldn't `channel_idx` always span the 2nd dim of `data`? Why not just have `data=data`?

Reply:
these channels_idx are a subset of the total channels in the raw dataset. : `len(channel_idx)<=data.shape[1]`

Reply:
oh, I see